src/package-request.js

```diff
@@ -132,21 +132,27 @@ export default class PackageRequest {
     }
   }
 
-  async normalizeRange(pattern: string): Promise<string> {
+  async normalizeRange(name: string, pattern: string): Promise<string> {
```

Comment:
>If I remember correctly we can delete every Promise.resolve because async/await world we should not use them (i.e. It automatically change the result in Promise.resolve).

Oh, right. This now makes it a double promise. Great catch!

>Do you think we can start to add some unit tests for package-request functions?

I'd love to but I'd also love to tie them to a "business" purpose. Right now most of these seem to be covered via integration tests. That said I want to do a pass on tests to make them more approachable so may be then?

>I am afraid that every function in this file is not try to adress one responsability and we couple the code in function just for code duplication. One another proof of this is that we have many similar names. If we can't name something, maybe there is a reason.

That's a good observation. I think we can give this file a closer look. That said I'd rather not try that in this patch since this PR resolves an important issue and refactoring those may create some back and forths.